ColorMyLogiXNodes/ColorMyLogiXNodes.cs

```diff
@@ -256,356 +257,389 @@ private enum InputNodeOverrideEnum
 		//	Sine
 		//}
 
+		// NodeInfo
 		private static NodeInfo nullNodeInfo = new();
+		private static HashSet<NodeInfo> nodeInfoSet = new();
 
+		// Random number generators
 		private static System.Random rng;
 		private static System.Random rngTimeSeeded = new System.Random();
 
+		// Tags
 		private const string COLOR_SET_TAG = "ColorMyLogiX.ColorSet";
 
-		private static HashSet<NodeInfo> nodeInfoSet = new();
-
+		// Sync reference Map
 		private static Dictionary<ISyncRef, IWorldElement> syncRefTargetMap = new();
 
+		// Threading
 		private static ManualResetEvent manualResetEvent = new(false);
-
 		private const int THREAD_INNER_SLEEP_TIME_MILLISECONDS = 0;
 
+		// Real-time color change
 		private const int REALTIME_COLOR_CHANGE_INTERVAL_MILLISECONDS = 200;
-
 		private const bool ALWAYS_THROTTLE_REALTIME_COLOR_CHANGE = false;
-
 		private static long lastColorChangeTime = DateTime.UtcNow.Ticks;
 
+
 		// have any logix visual customizer patches been found?
 		private static bool? lvcHasPatches = null;
 
 		public override void OnEngineInit()
 		{
-			Harmony harmony = new Harmony($"owo.{Author}.{Name}");
-			Config = GetConfiguration()!;
-			Config.Unset(USE_AUTO_RANDOM_COLOR_CHANGE);
-			Config.Save(true);
-			harmony.PatchAll();
-
-			nullNodeInfo.node = null;
-			nullNodeInfo.bgField = null;
-			nullNodeInfo.textFields = null;
-
-			Thread thread1 = new(new ThreadStart(RefDriverNodeThread));
-			thread1.Start();
-
-			Thread thread2 = new(new ThreadStart(StandardNodeThread));
-			thread2.Start();
+			InitializeHarmony();
+			InitializeNullNodeInfo();
+			StartRefDriverNodeThread();
+			StartStandardNodeThread();
 
 			Config.OnThisConfigurationChanged += (configChangedEvent) =>
 			{
-				if ((configChangedEvent.Key == MOD_ENABLED && !Config.GetValue(MOD_ENABLED)) || (configChangedEvent.Key == AUTO_UPDATE_REF_AND_DRIVER_NODES && !Config.GetValue(AUTO_UPDATE_REF_AND_DRIVER_NODES)))
-				{
-					Debug("syncRefTargetMap Size before clear: " + syncRefTargetMap.Count.ToString());
-					foreach (ISyncRef syncRef in syncRefTargetMap.Keys.ToList())
-					{
-						syncRefTargetMap[syncRef] = null;
-						syncRefTargetMap.Remove(syncRef);
-					}
-					syncRefTargetMap.Clear();
-					Debug("Cleared syncRefTargetMap. New size: " + syncRefTargetMap.Count.ToString());
-				}
+				bool modEnabled = Config.GetValue(MOD_ENABLED);
+				bool autoUpdateRefAndDriverNodes = Config.GetValue(AUTO_UPDATE_REF_AND_DRIVER_NODES);
+				bool updateNodesOnConfigChanged = Config.GetValue(UPDATE_NODES_ON_CONFIG_CHANGED);
+				bool useAutoRandomColorChange = Config.GetValue(USE_AUTO_RANDOM_COLOR_CHANGE);
 
-				if ((configChangedEvent.Key == MOD_ENABLED && !Config.GetValue(MOD_ENABLED)) || ((configChangedEvent.Key == UPDATE_NODES_ON_CONFIG_CHANGED || configChangedEvent.Key == USE_AUTO_RANDOM_COLOR_CHANGE && (!Config.GetValue(UPDATE_NODES_ON_CONFIG_CHANGED)) && !Config.GetValue(USE_AUTO_RANDOM_COLOR_CHANGE))))
+				if (!modEnabled || !autoUpdateRefAndDriverNodes)
```

Comment:
Should be checking configChangedEvent.Key here

Reply:
Added the check back

Reply:
It is still missing some of the checks from before. I actually don't think this behaves the same as my original code.